### ISSUE TITLE
ZIO HTTP Doesn't Respect Bash Naming Style (Environment Variables)

### ISSUE THREAD

**Issue by khajavi** [zio/zio-http#2803]
Assume I have the following ZIO HTTP App:

```scala
object MainApp extends ZIOAppDefault {
  def run =
    Server.serve(
      Routes(
        Method.GET / "hello" -> handler(Response.ok),
      ),
    ).provide(
      Server.live,
      ZLayer.fromZIO(ZIO.config(Server.Config.config)).debug)
    )
}
```

If of set:

```
export BINDING_PORT=3000
```

But it doesn't respect it and after debugging find out that it requires `BINDING-PORT`

The workaround is to map key names using `Server.Config.config.mapKey(_.replace('-', '_'))` which is not idiomatic.

**Comment by jdegoes** [comment]
/bounty $50

Not sure if the fix is here or ZIO core, though.

**Comment by 987Nabil** [comment]
There is a test for exactly this case in zio core. Interesting that it is failing here.

**Comment by 42njaber** [comment]
/attempt #2803

<div id="algora-attempt" />

<details>
  <summary>Options</summary>
  <ul>
    <li>
      <a href="https://console.algora.io/api/bounties/clx17kriy000hjl09d65yc7c0/cancel-attempt">
        Cancel my attempt
      </a>
    </li>
  </ul>
</details>

**Comment by 987Nabil** [comment]
Fixed in zio 2.1.0
